docs/customizing_token_claims.rst

```diff
@@ -15,23 +13,28 @@ generated by the ``TokenObtainPairView``:
   from rest_framework_simplejwt.views import TokenObtainPairView
 
   class MyTokenObtainPairSerializer(TokenObtainPairSerializer):
-      @classmethod
-      def get_token(cls, user):
-          token = super().get_token(user)
-
+      def validate(self, attrs):
+          token = super().validate(attrs)
+ 
           # Add custom claims
-          token['name'] = user.name
+          token['name'] = self.user.name
```

Comment:
You are chasing the physical payload rather than the internal payload of the token itself.